aspnetcore/release-notes/aspnetcore-5.0.md

```diff
@@ -0,0 +1,464 @@
+
+[OpenAPI Specification](http://spec.openapis.org/oas/v3.0.3) is an industry standard for describing HTTP APIs and integrating them into complex business processes or with third parties. OpenAPI is widely supported by all cloud providers and many API registries. Apps that emit OpenAPI documents from web APIs have a variety of new opportunities in which those APIs can be used. In partnership with the maintainers of the open-source project [Swashbuckle.AspNetCore](https://www.nuget.org/packages/Swashbuckle.AspNetCore/), the ASP.NET Core API template contains a NuGet dependency on [Swashbuckle](https://github.com/domaindrivendev/Swashbuckle.AspNetCore). Swashbuckle is a popular open-source NuGet package that emits OpenAPI documents dynamically. Swashbuckle does this by introspecting over the API controllers and generating the OpenAPI document at run-time, or at build time using the Swashbuckle CLI.
+
+In ASP.NET Core 5.0, the web API templates enable the OpenAPI support by default. To disable OpenAPI:
+
+* From the command line:
+
+	```dotnetcli
+	dotnet new webapi --no-openapi true
+	```
+* From Visual Studio: Uncheck **Enable OpenAPI support**.
+
+All *.csproj* files created for web API projects contain the [Swashbuckle.AspNetCore](https://www.nuget.org/packages/Swashbuckle.AspNetCore/) NuGet package reference.
+
+```xml
+<ItemGroup>
+    <PackageReference Include="Swashbuckle.AspNetCore" Version="5.5.1" />
+</ItemGroup>
+```
+
+The template generated code contains code in `Startup.ConfigureServices` that activates OpenAPI document generation:
+
+[!code-csharp[](~/release-notes/sample/StartupSwagger.cs?name=snippet)]
+
+The `Startup.Configure` method adds the Swashbuckle middleware, which enables the:
+
+* Document generation process.
+* Swagger UI page by default in development mode.
+
+The template generated code won't accidentally expose the API's description when publishing to production.
+
+[!code-csharp[](~/release-notes/sample/StartupSwagger.cs?name=snippet2)]
+
+#### Azure API Management Import
+
+When ASP.NET Core API projects enable OpenAPI, the Visual Studio 2019 version 16.8 and later publishing automatically offer an additional step in the publishing flow. Developers who use [Azure API Management](xref:tutorials/publish-to-azure-api-management-using-vs) have an opportunity to automatically import the APIs into Azure API Management during the publish flow:
+
+![Azure API Management Import VS publishing](~/release-notes/static/publish-to-apim.png)
+
+### Better launch experience for web API projects
+
+With OpenAPI enabled by default, the app launching experience (F5) for web API developers significantly improves. With .NET, the web API template comes pre-configured to load up the Swagger UI page. The Swagger UI page provides both the documentation added for the published API, and enables testing the APIs with a single click.
+
+![swagger/index.html view](~/release-notes/static/swagger-ui-page-rc1.png)
+
+## Blazor
```

Comment:
We did a bunch of work to significantly improve runtime performance in Blazor WebAssembly for .NET 5. Can we add a section to highlight that? Details here: https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-5-release-candidate-1/#blazor-webassembly-performance-improvements